testbed/integration/k8senrichment/e2e_test.go

```diff
@@ -0,0 +1,241 @@
+			attrs: map[string]*expectedValue{
+				"k8s.pod.name":                newExpectedValue(equal, "telemetrygen-"+testID+"-traces-statefulset-0"),
+				"k8s.pod.uid":                 newExpectedValue(regex, uidRe),
+				"k8s.statefulset.name":        newExpectedValue(equal, "telemetrygen-"+testID+"-traces-statefulset"),
+				"dt.kubernetes.workload.name": newExpectedValue(equal, "telemetrygen-"+testID+"-traces-statefulset"),
+				"dt.kubernetes.workload.kind": newExpectedValue(equal, "statefulset"),
+				"k8s.namespace.name":          newExpectedValue(equal, testNs),
+				"k8s.node.name":               newExpectedValue(exist, ""),
+				"k8s.cluster.uid":             newExpectedValue(regex, uidRe),
+				"dt.kubernetes.cluster.id":    newExpectedValue(regex, uidRe),
+			},
+		},
+		{
+			name:    "traces-deployment",
+			service: "test-traces-deployment",
+			attrs: map[string]*expectedValue{
+				"k8s.pod.name":                newExpectedValue(regex, "telemetrygen-"+testID+"-traces-deployment-[a-z0-9]*-[a-z0-9]*"),
+				"k8s.pod.uid":                 newExpectedValue(regex, uidRe),
+				"k8s.deployment.name":         newExpectedValue(equal, "telemetrygen-"+testID+"-traces-deployment"),
+				"dt.kubernetes.workload.name": newExpectedValue(equal, "telemetrygen-"+testID+"-traces-deployment"),
+				"dt.kubernetes.workload.kind": newExpectedValue(equal, "deployment"),
+				"k8s.namespace.name":          newExpectedValue(equal, testNs),
+				"k8s.node.name":               newExpectedValue(exist, ""),
+				"k8s.cluster.uid":             newExpectedValue(regex, uidRe),
+				"dt.kubernetes.cluster.id":    newExpectedValue(regex, uidRe),
+			},
+		},
+		{
+			name:    "traces-daemonset",
+			service: "test-traces-daemonset",
+			attrs: map[string]*expectedValue{
+				"k8s.pod.name":                newExpectedValue(regex, "telemetrygen-"+testID+"-traces-daemonset-[a-z0-9]*"),
+				"k8s.pod.uid":                 newExpectedValue(regex, uidRe),
+				"k8s.daemonset.name":          newExpectedValue(equal, "telemetrygen-"+testID+"-traces-daemonset"),
+				"dt.kubernetes.workload.name": newExpectedValue(equal, "telemetrygen-"+testID+"-traces-daemonset"),
+				"dt.kubernetes.workload.kind": newExpectedValue(equal, "daemonset"),
+				"k8s.namespace.name":          newExpectedValue(equal, testNs),
+				"k8s.node.name":               newExpectedValue(exist, ""),
+				"k8s.cluster.uid":             newExpectedValue(regex, uidRe),
+				"dt.kubernetes.cluster.id":    newExpectedValue(regex, uidRe),
+			},
+		},
+	}
+
+	for _, tc := range tcs {
+		scanTracesForAttributes(t, tracesConsumer, tc.service, tc.attrs)
+	}
+}
+
+func scanTracesForAttributes(t *testing.T, ts *consumertest.TracesSink, expectedService string,
+	kvs map[string]*expectedValue) {
+	// Iterate over the received set of traces starting from the most recent entries due to a bug in the processor:
+	// https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/18892
```

Comment:
I assume this was copied from upstream, but the issue has been resolved. Did the associated PR not fix this problem?

Reply:
Ah good find, I meant to remove this after the clean up but forgot. I will remove!